controllers/handler/node_controller.go

```diff
@@ -117,10 +118,13 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, request ctrl.Request) (c
 }
 
 func (r *NodeReconciler) getDependencyVersions() *nmstate.DependencyVersions {
-	handlerNetworkManagerVersion, err := nmstate.ExecuteCommand("NetworkManager", "--version")
+	handlerNetworkManagerVersion, err := nmstate.ExecuteCommand("nmcli", "--version")
 	if err != nil {
 		r.Log.Info("error retrieving handler NetworkManager version: %s", err.Error())
 	}
+	// remove leading characters up to last space
+	split := strings.Split(handlerNetworkManagerVersion, " ")
```

Comment:
Could you please check for length of the split before indexing it? To make sure that split has at least 1 element, and avoid indexing with negative number (which would panic)

Reply:
If string does not contain separator, Split returns a slice of length 1 whose only element is original string. So this check is not necessary, imo - it will always evaluate to true. Example: [https://go.dev/play/p/wcabqGIWJAr](https://go.dev/play/p/wcabqGIWJAr)